xbmc/input/InputManager.cpp

```diff
@@ -190,22 +190,42 @@ bool CInputManager::ProcessEventServer(int windowId, float frameTime)
   }
 
   // now handle any buttons or axis
-  std::string joystickName;
+  std::string strMapName;
   bool isAxis = false;
   float fAmount = 0.0;
+  bool isJoystick = false;
 
   // es->ExecuteNextAction() invalidates the ref to the CEventServer instance
   // when the action exits XBMC
   es = CEventServer::GetInstance();
   if (!es || !es->Running() || es->GetNumberOfClients() == 0)
     return false;
-  unsigned int wKeyID = es->GetButtonCode(joystickName, isAxis, fAmount);
+  unsigned int wKeyID = es->GetButtonCode(strMapName, isAxis, fAmount, isJoystick);
 
   if (wKeyID)
   {
-    if (joystickName.length() > 0)
+    if (strMapName.length() > 0)
     {
-      return false;
+      // joysticks ar not supported via eventserver
```

Comment:
small typo. and point of clarification - eventserver used to support two kinds of joystick, the first was a 360-style that used our deceased joystick system and the second is an original-xbox style that is handled directly below this code.

The original xbox style is still supported, it uses the [gamepad.xml](https://github.com/xbmc/xbmc/blob/master/system/keymaps/gamepad.xml) keymap so I guess it would be called "gamepad" instead of "joystick".  You might want to mention that joysticks are still supported, but as xbox-style "gamepads".